src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -2826,319 +2776,147 @@ void InitZoneAirSetPoints(EnergyPlusData &state)
         state.dataHeatBalFanSys->PreviousMeasuredHumRat3.dimension(NumOfZones, 0.0);
 
         // Allocate Derived Types
-        ZoneSysEnergyDemand.allocate(NumOfZones);
-        ZoneSysMoistureDemand.allocate(NumOfZones);
+        state.dataZoneEnergyDemand->ZoneSysEnergyDemand.allocate(NumOfZones);
+        state.dataZoneEnergyDemand->ZoneSysMoistureDemand.allocate(NumOfZones);
+        if (state.dataHeatBal->doSpaceHeatBalanceSimulation || state.dataHeatBal->doSpaceHeatBalanceSizing) {
+            state.dataZoneEnergyDemand->spaceSysEnergyDemand.allocate(state.dataGlobal->numSpaces);
+            state.dataZoneEnergyDemand->spaceSysMoistureDemand.allocate(state.dataGlobal->numSpaces);
+        }
 
-        for (Loop = 1; Loop <= NumOfZones; ++Loop) {
+        for (int zoneNum = 1; zoneNum <= NumOfZones; ++zoneNum) {
             FirstSurfFlag = true;
-            for (SurfNum = Zone(Loop).HTSurfaceFirst; SurfNum <= Zone(Loop).HTSurfaceLast; ++SurfNum) {
-                if (FirstSurfFlag) {
-                    TRefFlag = state.dataSurface->SurfTAirRef(SurfNum);
-                    FirstSurfFlag = false;
-                }
-                // for each particular zone, the reference air temperature(s) should be the same
-                // (either mean air, bulk air, or supply air temp).
-                if (state.dataSurface->SurfTAirRef(SurfNum) != TRefFlag) {
-                    ShowWarningError(state, "Different reference air temperatures for difference surfaces encountered in zone " + Zone(Loop).Name);
+            for (int spaceNum : state.dataHeatBal->Zone(zoneNum).spaceIndexes) {
+                auto &thisSpace = state.dataHeatBal->space(spaceNum);
+                for (int SurfNum = thisSpace.HTSurfaceFirst; SurfNum <= thisSpace.HTSurfaceLast; ++SurfNum) {
+                    if (FirstSurfFlag) {
+                        TRefFlag = state.dataSurface->SurfTAirRef(SurfNum);
+                        FirstSurfFlag = false;
+                    }
+                    // for each particular zone, the reference air temperature(s) should be the same
+                    // (either mean air, bulk air, or supply air temp).
+                    if (state.dataSurface->SurfTAirRef(SurfNum) != TRefFlag) {
+                        ShowWarningError(state,
+                                         "Different reference air temperatures for difference surfaces encountered in zone " +
+                                             state.dataHeatBal->Zone(zoneNum).Name);
+                    }
                 }
             }
         }
 
         // CurrentModuleObject='Zone'
-        for (Loop = 1; Loop <= NumOfZones; ++Loop) {
-            SetupOutputVariable(state,
```

Comment:
Moved to one of several new `setupOutputVars` methods depending on what struct holds the data (heat balance, demands, separate reporting struct, etc.).